site/src/scaffold/scripts/popover.ts

```diff
@@ -0,0 +1,72 @@
+export function initPopovers(container: HTMLElement) {
+	container.addEventListener('click', (event: Event) => {
+		let targetPopover: HTMLDetailsElement | null;
+		targetPopover =
+			event.target instanceof Element &&
+			(event.target.closest('details.popover') as HTMLDetailsElement)
+				? (event.target.closest('details.popover') as HTMLDetailsElement)
+				: null;
+
+		if (!targetPopover) {
+			targetPopover =
+				event.target instanceof Element &&
+				event.target.shadowRoot &&
+				event.target.shadowRoot.activeElement &&
+				(event.target.shadowRoot.activeElement.closest('details.popover') as HTMLDetailsElement)
+					? (event.target.shadowRoot.activeElement.closest('details.popover') as HTMLDetailsElement)
+					: null;
+		}
+
+		if (!targetPopover) {
+			return;
+		}
+
+		const allPopovers: HTMLDetailsElement[] = Array.from(
+			container.querySelectorAll('details.popover')
+		);
+
+		if (!targetPopover.open) {
+			const keyHandler = (event: KeyboardEvent) => {
+				if (event.key === 'Escape') {
+					closePopovers();
+				}
+			};
+
+			const clickHandler = (event: Event) => {
+				if (!(event.target instanceof Element)) {
+					return;
+				}
+
+				if (!targetPopover?.contains(event.target)) {
+					targetPopover ? closePopovers([targetPopover]) : closePopovers();
+				}
+			};
+
+			const blurHandler = () => {
+				if ((document.activeElement?.nodeName || '').toLowerCase() === 'iframe') {
```

Comment:
I think we can remove the need for the `|| ''` fallback by continuing optional chaining on the `toLowerCase` method:

```suggestion
				if (document.activeElement?.nodeName?.toLowerCase() === 'iframe') {
```

Thoughts?

Reply:
Updated